engines/advancedDetector.h

```diff
@@ -481,10 +600,27 @@ class AdvancedMetaEngineDetection : public MetaEngineDetection {
 	friend class FileMapArchive;
 };
 
+template<class Descriptor>
+class AdvancedMetaEngineDetection : public AdvancedMetaEngineDetectionBase {
+protected:
+	AdvancedMetaEngineDetection(const Descriptor *descs, const PlainGameDescriptor *gameIds) : AdvancedMetaEngineDetectionBase(descs, sizeof(Descriptor), gameIds) {}
+
+	Common::Error identifyGame(DetectedGame &game, const void **descriptor) override {
+		Common::Error err = AdvancedMetaEngineDetectionBase::identifyGame(game, descriptor);
+		if (err.getCode() != Common::kNoError) {
+			return err;
+		}
+		if (*descriptor) {
```

Comment:
No, `descriptor` is always supposed to be non-null (I will add an assert for it).
What I check here is if `AdvancedMetaEngineDetectionBase::identifyGame` generated a descriptor (that should always been verified) to avoid transforming a nullptr into a valid pointer.

Reply:
This descriptor is what was causing us a lot of troubles: it points in static data section of detection plugin.
I didn't see any engine allocating it dynamically.
If it does, then it should not inherit from this class but directly from the AdvancedMetaEngineBase I think.

Reply:
I think I understand. Could you add a comment explaining that the code is replacing a pointer to static memory with a pointer to dynamically allocated memory?